build.gradle

```diff
@@ -3,26 +3,329 @@ buildscript {
+    group = 'Install'
+    description = 'Install the artifacts of Realm project to the Maven Repo in ANDROID_HOME'
+    buildFile = file('realm/build.gradle')
+    tasks = ['uploadArchives']
+}
+
+task assembleExamples(type:GradleBuild) {
+    dependsOn installGradlePlugin
+    dependsOn installRealm
+    group = 'Build'
+    description = 'Assemble the Realm examples'
+    buildFile = file('examples/build.gradle')
+    tasks = ['assemble']
+}
+
+task monkeyExamples(type:GradleBuild) {
+    dependsOn installGradlePlugin
+    dependsOn installRealm
+    group = 'Build'
+    description = 'Run the monkey tests on the Realm examples'
+    buildFile = file('examples/build.gradle')
+    tasks = ['monkeyRelease']
+}
+
+task javadoc(type:GradleBuild) {
+    description = 'Generate the Javadoc Jar for the Realm project'
+    group = 'Docs'
+    buildFile = file('realm/build.gradle')
+    tasks = ['javadocJar']
+}
+
+task eclipseJar(type:GradleBuild) {
+    description = 'Generate the Jar file to be used with Eclipse'
+    group = 'Artifact'
+    buildFile = file('realm/build.gradle')
+    tasks = ['shadowJar']
+}
+
+task distributionPackage(type:Zip) {
+    description = 'Generate the distribution package'
+    dependsOn eclipseJar
+    dependsOn assembleRealm
+    dependsOn javadoc
+
+    group = 'Artifact'
+    archiveName = "realm-java-${currentVersion}.zip"
+    destinationDir = file("${buildDir}/outputs/distribution")
+
+    from('changelog.txt')
+    from('LICENSE')
+    from('version.txt')
+    from('realm/realm-annotations-processor/build/libs') {
+        include 'realm-android-*.jar'
+        into 'eclipse'
+    }
+    from('realm/realm-library/src/main/jniLibs/') {
+        include '**/*'
+        into 'eclipse'
+    }
+    from('realm/realm-library/build/libs') {
+        include 'realm-android-*-javadoc.jar'
+        into 'docs'
+    }
+    from('realm/realm-library/build/docs') {
+        include '**/*'
+        into 'docs'
+    }
+    from('examples') {
+        exclude 'local.properties'
+        exclude '**/gradle'
+        exclude '**/.gradle'
+        exclude '**/build'
+        into 'examples'
+    }
+}
+
+task cleanRealm(type:GradleBuild) {
+    description = 'Clean the Realm project'
+    group = 'Clean'
+    buildFile = file('realm/build.gradle')
+    tasks = ['clean']
+}
+
+task cleanGradlePlugin(type:GradleBuild) {
+    description = 'Clean the Realm Gradle plugin project'
+    group = 'Clean'
+    buildFile = file('gradle-plugin/build.gradle')
+    tasks = ['clean']
+}
+
+task cleanExamples(type:GradleBuild) {
+    description = 'Clean the Realm examples'
+    group = 'Clean'
+    buildFile = file('examples/build.gradle')
+    tasks = ['clean']
+}
+
+task cleanLocalMavenRepos(type:Delete) {
+    description = 'Remove any Realm artifacts from the local Maven repositories'
+    group = 'Clean'
+    dependsOn checkEnvironment
+    delete "${System.env.ANDROID_HOME}/extras/android/m2repository/io/realm"
```

Comment:
never mind I was referring to the variable you created inside `checkEnvironment`